bedwars-plugin/src/main/java/com/andrei1058/bedwars/money/internal/MoneyPerMinuteTask.java

```diff
@@ -22,10 +22,12 @@ public MoneyPerMinuteTask(Arena arena) {
         if (money < 1){
             return;
         }
-        task = Bukkit.getScheduler().runTaskTimer( BedWars.plugin, () -> {
-            for (Player p : arena.getPlayers()) {
-                    BedWars.getEconomy ().giveMoney ( p, money );
-                    p.sendMessage ( Language.getMsg ( p, Messages.MONEY_REWARD_PER_MINUTE ).replace ( "{money}", String.valueOf ( money ) ) );
+        task = Bukkit.getScheduler().runTaskTimerAsynchronously(BedWars.plugin, () -> {
```

Comment:
This is not thread safe man. just implement the NPE checker on a sync thread..

Reply:
How is it not thread safe?

Reply:
you're iterating arena#players async. why you making everything async on your pull requests? try googling concurrent programming

Reply:
Maybe because I’ve heard too much about performance being bad so moving things off thread help that